test/download.cpp

```diff
@@ -0,0 +1,80 @@
+#include "download.h"
+
+#include "test-helpers.h"
+
+using namespace podboat;
+
+TEST_CASE("Require-view-update callback gets called when download progress or status changes",
+	"[Download]")
+{
+	GIVEN("A Download object") {
+		bool got_called = false;
+		std::function<void()> callback = [&got_called]() {
+			got_called = true;
+		};
+
+		Download d(callback);
+
+		REQUIRE(d.status() == DlStatus::QUEUED);
+
+		WHEN("its progress is updated (increased)") {
+			got_called = false;
```

Comment:
There is no need to reset the variable: Catch2 re-runs the whole TEST_CASE in such a way that each given-when-then path is run exactly once. Same goes for `SECTION` macros: TEST_CASE is re-run such that each "leaf" section is visited only once.

Ditto for other resets below.